test/unit/color.js

```diff
@@ -133,187 +133,109 @@
-    assert.equal(oColor.toHex(), '180637');
-  });
-
-  QUnit.test('fromHsl (with whitespaces)', function(assert) {
-    assert.ok(typeof fabric.Color.fromHsl === 'function');
-    var originalHsl = 'hsl( 262 , 80% , 12% )';
-    var oColor = fabric.Color.fromHsl(originalHsl);
-    assert.ok(oColor);
-    assert.ok(oColor instanceof fabric.Color);
-    assert.equal(oColor.toHsl(), 'hsl(262,80%,12%)');
-    assert.equal(oColor.toHex(), '180637');
+  QUnit.module('parsing colors');
+  [{
+    name: 'fromRgb',
+    stringToParse: 'rgb(255,255,255)',
+    expectedSource: [255, 255, 255, 1]
+  },{
+    name: 'fromRgb no commas',
+    stringToParse: 'rgb(255 0 255)',
+    expectedSource: [255, 0, 255, 1]
+  },{
+    name: 'fromRgb (with whitespaces)',
+    stringToParse: 'rgb( 255 , 128 , 64 )',
+    expectedSource: [255, 128, 64, 1]
+  },{
+    name: 'fromRgb no commas (with whitespaces)',
+    stringToParse: 'rgb( 255    128 64 )',
+    expectedSource: [255, 128, 64, 1]
+  },{
+    name: 'fromRgb (percentage values)',
+    stringToParse: 'rgb(100%,50%,25%)',
+    expectedSource: [255, 127, 64, 1]
+  },{
+    name: 'fromRgb (percentage values with whitespaces)',
+    stringToParse: 'rgb(100% ,   50% ,  25%)',
+    expectedSource: [255, 127, 64, 1]
+  },{
+    name: 'fromRgba',
+    stringToParse: 'rgba(255,12,10,0.5)',
+    expectedSource: [255, 12, 10, 0.5]
+  },{
+    name: 'fromRgba without commas',
+    stringToParse: 'rgba(255 12 10 / 0.5)',
+    expectedSource: [255, 12, 10, 0.5]
+  },{
+    name: 'fromRgba (with spaces and missing 0)',
+    stringToParse: 'rgba( 255 , 12 , 10 , .3 )',
+    expectedSource: [255, 12, 10, 0.3]
+  },{
+    name: 'fromRgba (with whitespaces)',
+    stringToParse: 'rgba( 255 , 33 , 44 , 0.6 )',
+    expectedSource: [255, 33, 44, 0.6]
+  },{
+    name: 'fromRgba (percentage values)',
+    stringToParse: 'rgba(100%,50%,25%,33%)',
+    expectedSource: [255, 127, 64, 0.33]
+  },{
+    name: 'fromRgba (percentage values)',
+    stringToParse: 'rgba(  100.00%  ,50.40%,   25.1%   ,  33%  )',
+    expectedSource: [255, 129, 64, 0.33]
+  },{
+    name: 'fromRgba (percentage values with whitespaces)',
+    stringToParse: 'rgba(  100.00%  ,50.80%,   25.1%   ,  33%  )',
+    expectedSource: [255, 130, 64, 0.33]
+  },{
+    name: 'fromRgba (percentage values with whitespaces no zeroes)',
+    stringToParse: 'rgba(  .99%  ,50.40%,   25.1%   ,  .33  )',
+    expectedSource: [3, 129, 64, 0.33]
+  }].forEach(({ name, stringToParse, expectedSource }) => {
+    QUnit.test(name, function(assert) {
+      var oColor = fabric.Color.fromRgba(stringToParse);
+      assert.ok(oColor);
+      assert.ok(oColor instanceof fabric.Color);
+      assert.deepEqual(oColor.getSource(), expectedSource);
+      var oColorUppercase = fabric.Color.fromRgb(stringToParse.toUpperCase());
+      assert.ok(oColorUppercase);
+      assert.ok(oColorUppercase instanceof fabric.Color);
+      assert.deepEqual(oColorUppercase.getSource(), expectedSource);
```

Comment:
seems this function can be reused in the hsl loop as well

Reply:
could but i didn't becuase  `fromRgb` vs `fromHsl` and didn't want to overcomplicate it. Still better than all different tests.

Reply:
thought of 
```
function runColorTest({ name, stringToParse, expectedSource }, from, fromAlpha) {
      var oColor = fromAlpha(stringToParse);
      assert.ok(oColor);
      assert.ok(oColor instanceof fabric.Color);
      assert.deepEqual(oColor.getSource(), expectedSource);
      var oColorUppercase = from(stringToParse.toUpperCase());
      assert.ok(oColorUppercase);
      assert.ok(oColorUppercase instanceof fabric.Color);
      assert.deepEqual(oColorUppercase.getSource(), expectedSource);
}

```